src/transactions/test/PathPaymentTests.cpp

```diff
@@ -558,65 +558,80 @@ TEST_CASE("pathpayment", "[tx][pathpayment]")
         });
     }
 
-    SECTION("path payment send issuer missing")
+    SECTION("issuer missing")
     {
-        auto market = TestMarket{*app};
-        auto source = root.create("source", minBalance1);
-        auto destination = root.create("destination", minBalance1);
-        source.changeTrust(idr, 20);
-        destination.changeTrust(usd, 20);
-        gateway.pay(source, idr, 10);
-        for_all_versions(*app, [&] {
-            gateway.merge(root);
-            REQUIRE_THROWS_AS(
-                source.pay(destination, idr, 11, usd, 11, {}, &idr),
-                ex_PATH_PAYMENT_STRICT_RECEIVE_NO_ISSUER);
-            // clang-format off
-            market.requireBalances(
-                {{source, {{xlm, minBalance1 - 2 * txfee}, {idr, 10}, {usd, 0}}},
-                 {destination, {{xlm, minBalance1 - txfee}, {idr, 0}, {usd, 0}}}});
-            // clang-format on
-        });
-    }
+        // look at the SECTION "path payment uses all offers in a loop" for a
+        // successful path payment test with no issuers
 
-    SECTION("path payment middle issuer missing")
-    {
         auto market = TestMarket{*app};
         auto source = root.create("source", minBalance1);
         auto destination = root.create("destination", minBalance1);
         source.changeTrust(idr, 20);
         destination.changeTrust(usd, 20);
         gateway.pay(source, idr, 10);
-        for_all_versions(*app, [&] {
-            auto btc = makeAsset(getAccount("missing"), "BTC");
-            REQUIRE_THROWS_AS(
-                source.pay(destination, idr, 11, usd, 11, {btc}, &btc),
-                ex_PATH_PAYMENT_STRICT_RECEIVE_NO_ISSUER);
-            // clang-format off
-            market.requireBalances(
-                {{source, {{xlm, minBalance1 - 2 * txfee}, {idr, 10}, {usd, 0}}},
-                 {destination, {{xlm, minBalance1 - txfee}, {idr, 0}, {usd, 0}}}});
-            // clang-format on
-        });
-    }
 
-    SECTION("path payment last issuer missing")
-    {
-        auto market = TestMarket{*app};
-        auto source = root.create("source", minBalance1);
-        auto destination = root.create("destination", minBalance1);
-        source.changeTrust(idr, 20);
-        destination.changeTrust(usd, 20);
-        gateway.pay(source, idr, 10);
         for_all_versions(*app, [&] {
-            gateway2.merge(root);
-            REQUIRE_THROWS_AS(
-                source.pay(destination, idr, 11, usd, 11, {}, &usd),
-                ex_PATH_PAYMENT_STRICT_RECEIVE_NO_ISSUER);
+            uint32_t ledgerVersion;
+            {
+                LedgerTxn ltx(app->getLedgerTxnRoot());
+                ledgerVersion = ltx.loadHeader().current().ledgerVersion;
+            }
+
+            SECTION("path payment send issuer missing")
+            {
+                gateway.merge(root);
+                if (ledgerVersion < 13)
```

Comment:
Same exact comments apply in this test: "if-else" repeated and formatting altered.